Redesigning lmms.io

Umcaruje opened the pull request:
**This is a WIP PR, please don't merge**

So, I wanted to redesign the page for quite a long time now, and as my entrance exam is over, I have more time for this issue. I wanted to open this PR so early, to keep the discussion about the redesign in one place. Note that atm only a part of the home page is done and the other pages are broken.

You can see a live version of the new redesign at http://lmms.io:8008/

Screenshot (can become outdated): 
![screenshot from 2016-06-29 23 03 41](https://cloud.githubusercontent.com/assets/6282045/16470180/00203f48-3e55-11e6-818f-a6af5a5bcb72.png)

I'm tagging @RebeccaDeField, since she's the designer of the new theme, and her input would be very valuable.

And as always, opinions and feedback are welcome.

tresf commented:
Here's a side-by-side:

![screen shot 2016-06-29 at 6 47 35 pm](https://cloud.githubusercontent.com/assets/6345473/16471332/009db07a-3e2a-11e6-843a-779df8217929.png)

I shared my opinions with @Umcaruje privately already but here they are publicly as well:
* I've never been a fan of the over-sized business-card style landing pages as I feel they're a waste of page real estate.  That said, I think the layout is sharp and the page is responsive, but I'm still not a fan of web pages putting fashion before function and that's how I feel about the extra-extra-large jumbotron style landing pages.  I'm probably a minority on this, but as the designer of the page we have today, I'd like to share my opinion.
* I feel the [background texture for the banner is distracting](https://cloud.githubusercontent.com/assets/6345473/16471474/ea35d67c-3e2a-11e6-80e3-c7b5fd65d502.png) and would be just as good as a plain color or perhaps a subtle gradient.  I also feel it clashes with the flatness of the software theme.  I didn't like it on the splash either, so if we're aiming for consistency, I'd rather it be in neither places.
* The [picture background I feel equally as distracted viewing](https://cloud.githubusercontent.com/assets/6345473/16471485/fecea6d6-3e2a-11e6-9533-f730acf9b98c.png).  I don't think the sliding-window effect is a good UI for a web page and if we keep it, I don't think the angle helps things.


That said, it's very nice to see a facelift to the webpage.  Editing CSS/HTML layout is a painstaking process and this is coming along very nicely.

Umcaruje commented:
Just an update, other pages now look better (added containers back) also made a macro so they have headers:
![screenshot from 2016-06-30 01 11 34](https://cloud.githubusercontent.com/assets/6282045/16471832/befc053c-3e5f-11e6-9299-517c01d9f560.png)

Spekular commented:
While I love the look of the huge new jumbotron, I have to agree with
@tresf that it decreases functionality. On a desktop things like that are
lovely, but on a weak laptop with a low res screen not so much.

I don't think either background texture is too distracting, and I probably
like the sliding effect (not sure how to test this PR so I haven't seen it
in action).

My own critique would be that the laptop is a bit awkward, especially if
it's going to slide. I'm sure there's a reason you removed the old video
box, but it was more functional and completely platform agnostic. The
border on this laptop will either be mac-like or not mac-like.

Spekular commented:
Apparently I couldn't read earlier, but now that I've checked out the website I've got a few critiques of my own.

- The laptop image took a while to load and just "scanned in" from top to bottom. Perhaps we could hide it until it's loaded and transition it in?
- The toolbar BG should ideally scroll counter to the page so it lines up with the other textured background instead of tearing, but I'm not sure how possible that is.
- "Get involved" lines up vertically between the top and bottom of the download arrow. Compared to the text it seems to sit a bit high.
- That transition from green and white to black and white and orange is a bit odd/abrupt. On their own both palettes are nice but I'm not 100% about the mix.
- The down arrow for screenshots doesn't change to an up arrow when pressed. In addition, I could only open and close this section once, after that the button did nothing.

Here's the tearing I was talking about:
![Tearing toolbar](https://cloud.githubusercontent.com/assets/8918426/16479030/e61b83de-3e9d-11e6-9d10-5ed79efbb146.png)

Here's a mockup I made of my earlier idea for a smaller banner:
![mockup of lmmsio](https://cloud.githubusercontent.com/assets/8918426/16479626/d61b7684-3ea1-11e6-815f-c4129daedbf5.png)

Umcaruje commented:
>On a desktop things like that are lovely, but on a weak laptop with a low res screen not so much.

The height of the jumbotron is calculated by the window height, and it takes 60% of the viewport, so it won't look huge on laptops, it'll just be smaller.

>The laptop image took a while to load and just "scanned in" from top to bottom. Perhaps we could hide it until it's loaded and transition it in?

The image is not final at the moment and it needs some optimising. I'll look into that.

>"Get involved" lines up vertically between the top and bottom of the download arrow. Compared to the text it seems to sit a bit high.

Good catch, I'll look into that.

>The down arrow for screenshots doesn't change to an up arrow when pressed. In addition, I could only open and close this section once, after that the button did nothing.

That is an unfinished part of the home page and will most likely get replaced with something else.

>The toolbar BG should ideally scroll counter to the page so it lines up with the other textured background instead of tearing, but I'm not sure how possible that is.

The background is lined up when you scroll to the top of the page. The scrolling background could be done perhaps, but it'd require javascript and would be an overkill or just look weird while scrolling up.

***

I have to say that I don't understand this hate towards a bigger header image. How does this affect functionality? The goal of the header is to actually interest the user and make him scroll the page. The current header just doesn't cut it for me. 

Horizontal space isn't generally something that's limiting in web pages. The site is responsive and users can scroll it on any device. the navbar is always there for them now, since it reappears when scrolling up, so there is no worry that people can't navigate the pages because the site is too long.

tresf commented:
> I have to say that I don't understand this hate towards a bigger header image. How does this affect functionality? The goal of the header is to actually interest the user and make him scroll the page. The current header just doesn't cut it for me.

Hate is a strong word in this context, but it's a matter of design opinion.

For example here's Paypal's US landing page:

![image](https://cloud.githubusercontent.com/assets/6345473/16494373/19fa1da0-3eb7-11e6-8bd6-b2f34dfb677b.png)

and here's WebEx's US landing page:

![image](https://cloud.githubusercontent.com/assets/6345473/16494405/337311b0-3eb7-11e6-8a3f-f69bfcfca7c9.png)

I find the latter to be intrusive, personally, despite it's popularity in modern site design.

Umcaruje commented:
@Spekular optimised the page, so it should load balizingly fast now. Also aligned the buttons: 
![screenshot from 2016-07-01 20 22 07](https://cloud.githubusercontent.com/assets/6282045/16530697/a37041c8-3fc9-11e6-9772-2bf30f1f6d35.png)
![screenshot from 2016-07-01 20 20 42](https://cloud.githubusercontent.com/assets/6282045/16530698/a374f2ae-3fc9-11e6-9c7d-62bb89f141f2.png)

Spekular commented:
Still got scanning on the image, although it seemed much faster. Maybe something else is causing it?

Brotli is probably not applicable here, I wrongly assumed it would be similar to optipng or tinypng.
As for optipng vs tinypng, tinypng is far smaller since its lossless. Color is slightly off on nearly every pixel (worse on a few), but I can't see any difference by eye.

Perhaps you already knew all of this, but whatever.

RebeccaDeField commented:
Here are some of the colors that I use for the ModernNoir theme that I think go very nicely with the green. You could use some of these colors instead of the gold if you want to.

![colors](https://cloud.githubusercontent.com/assets/7960169/16709308/c833449e-45c1-11e6-89a1-65850d28e38d.png)

Umcaruje commented:
Quick update: I switched to using SCSS for the website, which allows a lot more flexibility and usage of variables, which is crucial for the color work.

teeberg commented:
Good call! CSS without preprocessors is a hassle :-)

Umcaruje commented:
Update:
![screenshot from 2016-08-04 02 21 21](https://cloud.githubusercontent.com/assets/6282045/17386343/35c3e5b8-59ea-11e6-99ec-8aced99b4516.png)

Added a segment about how LMMS is open source, plus removed the screenshot portion and replaced it with an moving video background. The button will open a modal with the promo video, I have yet to implement that.

Removed the lightbox and everything associated with it.

I am also thinking about removing the community page, as soundcloud groups are getting deprecated , the facebook feed is broken, we rarely upload stuff to our YouTube, and google plus is dead as well.

tresf commented:
Yeah please kill the community pages.  We can always chercherry-pick from the archive if we ever want to resurrect a particular feed.  It'll improve navigation too, we can just provide external community links directly in the menu.

tresf commented on public/community.php:
```diff
@@ -1,404 +0,0 @@
-<?php include('header.php'); ?>
```
Goodbye, old friend. 😆

QrchackOfficial commented:
![2016-12-17 15_36_05-lmms _ home](https://cloud.githubusercontent.com/assets/3584315/21287556/8a16968a-c46e-11e6-99cc-a4c57efbc773.png)

> **macOS** (previously **Mac OS X**, then simply **OS X**) is the current series of Unix-based graphical operating systems developed and marketed by Apple Inc. designed to run on Apple's Macintosh computers, having been preinstalled on all Macs since 2002

It was never called "Apple OS X", can we at least use the right name? It's like saying LMMS is available for Ubuntu Linux.

Also, this last section:

![2016-12-17 15_38_22-lmms _ home](https://cloud.githubusercontent.com/assets/3584315/21287578/de2b2a2e-c46e-11e6-8e8c-1d4de604b81d.png)

The NetworkRedux banner is not visible at all. [This logo](https://networkredux.com/wp-content/uploads/2016/05/NR.png) should fit way better with the dark gray background. Also, the buttons feel... kinda out of place. Maybe center them vertically?

Umcaruje commented:
>The NetworkRedux banner is not visible at all. This logo should fit way better with the dark gray background. Also, the buttons feel... kinda out of place. Maybe center them vertically?

Hi, thanks for the feedback. The footer is not done yet, my classes started so I didn't have much time to finalize this redesign. So yeah I am planing on making all these changes.

Thanks for the macOS remark, I'll fix that

Umcaruje commented:
Idea: 
![image](https://cloud.githubusercontent.com/assets/6282045/21290541/8bc72580-c4be-11e6-8273-5bdb9efdb288.png)

Spekular commented:
I think the links and categories need to be reorganized somehow, but it's a difficult task. I'm thinking something along the lines of:

  * Elevator Pitch
    * Banner: Reduce height, add read more*
    * Features: "Making beats has never been easier", leave as is.
  * Various community involvement links:
    * Support: Forum, Documentation
    * Community: Discord, LSP, Soundcloud**
    * Development: Get involved, Github
  * Sales Pitch Redux (IE "not convinced yet?")
    * Promo Video
    * Download
    * Extensive feature list (as one section rather than three)

\* Something like this. Could scroll past banner when clicked or be static prompt.
![image](https://cloud.githubusercontent.com/assets/8918426/22929266/f9f94414-f2ba-11e6-8fe7-0f47dae43a03.png)


**I think an updated LMMS Service account would work well as a Soundcloud presence.
  * Post announcements as empty tracks or V/O tracks
  * Repost high quality music created in LMMS.
  * Create competition playlists w/ all competition entries, and with winners.
  * Post community collabs (eg. Avicii X You) or demo tracks?
  * Have a master playlist of everything LMMS produced, added to by SC message request?

---

This layout means there will be one download link at the top, one near the bottom, and one in the header. The community links also show up a tad earlier, which I think gives a good impression.

I think a sitemap with links to every page on lmms.io as well as external sites would be nice as well. This could be a bulleted list (bullets being page icons) that can elaborate more on what each site is good for. For example, Discord is used for sharing, dev chat, support, *and* casual conversation, and the forums have posts asking for help, sharing music, or posting tutorials. Mockup thing:

![image](https://cloud.githubusercontent.com/assets/8918426/22930163/3f919df6-f2bf-11e6-89bb-55537fd92b52.png)

Umcaruje commented:
Updates: I squashed the commits so it doesn't spam the comment section whenever I rebase. 
![image](https://cloud.githubusercontent.com/assets/6282045/24076533/692a0d96-0c33-11e7-89d2-47b3f97567b1.png)

The footer actually looks like something now, but I need feedback on it. I kinda fell like something else could be added to it.

Next on my list is the modal for the promo video and freshening up the LSP.

tresf commented:
I don't care for the white on green for the higher part, but in the footer, the white on dark gray looks great.  I especially like the consolidation of the social media buttons.
![screen shot 2017-03-18 at 6 59 21 pm](https://cloud.githubusercontent.com/assets/6345473/24076670/03c4101c-0c0d-11e7-924a-c7e9f2d82faf.png)

QrchackOfficial commented:
How about keeping the green LMMS branding across the whole thing?

Before:

![image](https://cloud.githubusercontent.com/assets/3584315/26766568/b3f33084-4994-11e7-873b-ff6c535c332d.png)

After:

![image](https://cloud.githubusercontent.com/assets/3584315/26766564/a4c2d4de-4994-11e7-96c1-d0c754dd73cd.png)

It would require
```css
a.stack {
  color: #10b146;
}
```

... possibly also change the hover color to black, like the top menu bar?

liushuyu commented:
Updated with changes from `master` branch
Just hoping I was not messing up with something or fuck up the whole branch

For the record:
https://github.com/liushuyu/lmms.io/tree/testing-old

Umcaruje commented:
@QrchackOfficial I believe that would make the whole website pretty dull and uninteresting. Also the orange has been implemented in the 1.2 theme by @RebeccaLaVie

liushuyu commented:
i18n strings are available on Transifex for translation: https://www.transifex.com/lmms/lmms/lmmsio/

RebeccaLaVie commented:
These headings are a little hard to read. The opacity of the pattern needs to be turned down.

![shot-2017-07-01_09-12-08](https://user-images.githubusercontent.com/7960169/27763705-a6318578-5e3d-11e7-90d4-3869594a7fe9.jpg)

Umcaruje commented:
![image](https://user-images.githubusercontent.com/6282045/31321159-8a182bfc-ac81-11e7-98c5-f25684ced562.png)
@RebeccaLaVie

RebeccaLaVie commented:
@Umcaruje The contrast looks sufficient to me. Thank you.

Umcaruje commented:
I have fixed everything I wanted on the redesign, and this is now officially ready for merge. 

Given this is a big merge, I'll give 48 hours of time for people to do final testing. I find the website redesign is stable and any other changes can be done at a later time (like merging competitions and showcase) :tada:

Umcaruje commented:
Ok, all looks good, I'm merging before time so I'm available to fix any problems that might occur from the change.
